### DISCUSSION TITLE
kotlinx.Serialization support to voyger in line with java.io.Serializable and Parcelable

### DISCUSSION THREAD

**Discussion by amitgcse** [adrielcafe/voyager#495] [General]
I would wish to use kotlinx's Serializable annotation to make my objects (and Screens) savable and recoverable.
I understand that now it works well with java.io.Serialzable and Parcelable. 
Example:
To be used in 
`class PostDetailsScreen(val post: Post): Screen{...}`

// Doesnot work;  (Can we make it work in future)
`@Serializable data class Post(val title: String)`

// Works 
`data class Post(val title: String): java.io.Serialzable`

**Comment by orcchg** [comment]
Same concern. The Serialization support has been added [here](https://github.com/adrielcafe/voyager/commit/043f53749cf362e61db4cf535d62c705527b30a9) a while ago. 

Sorry for a stupid question, but is there any option to disable custom `Screen` from saving/recover state, i.e. to prevent from serialization?

If is correct to have not-serializable properties while implementing custom `Screen`? i.e, passing lambdas or not-serializable classes as `val` properties into custom `Screen` ctor for the sake of dependency injection, configuration, callbacks, etc. Should we only pass serializable objects and primitives as parameters to custom `Screen`s?
